### PR TITLE
Support mixed updates

### PR DESCRIPTION
**Pull request by Garfield96**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -23,69 +23,71 @@
 
 using namespace std;
 
-// Reads edge list with comma separator
-pair<vector<pair<int, int>>, int> read_input(string filename, char delimiter = ',') {
+enum class Operation { READ, ADD, DELETE };
+
+// Reads edge list with separator
+pair<vector<tuple<Operation, int, int>>, int> read_mixed_input(string filename, Operation defaultOp) {
```
Comment (domargan) [inline src/main.cpp]: The name `read_mixed_input` could be a bit misleading in cases where we are just reading a core graph, which is never mixed.

Comment (domargan) [reply src/main.cpp]: Just to clarify, the updates are the only ones that are mixed and it would be good to distinguish that in the code.

**Review by domargan — CHANGES_REQUESTED** [review]
Apart from the tiny naming issue, all looks great. Please change it and I'll merge.

File: src/main.cpp (hunk 1 above)
Comment (domargan) [reply src/main.cpp]: Thus, just `read_input` would be a good option without the need to have separate functions.

Comment (Garfield96) [reply src/main.cpp]: I renamed the function to `read_input`

**Review by domargan — APPROVED** [review]

**Pull request merged**